### PR TITLE
[micromega] reformating using ocamlformat

### PR DESCRIPTION
**Pull request by fajb**

### CODE REVIEW

**Comment by fajb** [comment]
@maximedenes this is the PR with only ocamlformat.

**Review by maximedenes — APPROVED** [review]
LGTM. I personally don't like some of the changes (e.g. replacing `begin` ... `end` blocks by parentheses), but I don't care much.

**Comment by maximedenes** [comment]
@ejgallego please say if this looks good to you. Merging it will make it easier to review the other micromega PRs.

**Comment by ejgallego** [comment]
> @ejgallego please say if this looks good to you. Merging it will make it easier to review the other micromega PRs.

Indeed I was going to suggest to split the ocamlformat changes to its own PR; I think this is fine, maybe we can tweak the set of options to our liking.

When I mentioned `ocamlformat` in the original PR I did not intend to ask for a reformatting of the whole codebase; I first did suggest `ocp-indent` which works at a local way and only for indentation. But now that we are here maybe micromega can help as experiment to see how well do we like using that.

However before merging we need to add the `.ocamlformat` file to the repos and enforce it using dune support, [so that options are documented] I can take care of that quickly tho.

**Comment by ejgallego** [comment]
Hi folks, I've pushed a minimal config for `ocamlformat` and an update to docker files, a few comments:
- ocamlformat is still a bit unstable, in particular the defaults changed from 0.11 [which Frédéric used] to 0.12, the current version. I've setup a few options but still the resulting formatting is not the identity, it should be easy to do.
- @maximedenes , `begin/end` are now preserved in 0.12 , so if we run again the tool all these parenthesis should not be inserted.
- /all: there are many choices regarding formatting, IMHO the current choices [as done in this PR] are not exactly the style we do prefer, so I'd suggest everybody does play with the settings until we get something we like.

I think I'm gonna do a PR that just adds the config so people can play with it easily.

**Comment by maximedenes** [comment]
@ejgallego can I merge this? We have spurious changes in all micromega-related PRs (such as #11263), I'd like to make progress quickly (as in today).

**Comment by ejgallego** [comment]
IMO we need to merge #11257 first, then this.

**Comment by ejgallego** [comment]
> IMO we need to merge #11257 first, then this.

Note that there are minor differences with the formatting setup used in this PR and the setup in #11257 , due to improved functionality of the new version.

For example if you apply the setup of #11257 `begin` / `end` will be preserved, etc...

Let's discuss more on Gitter.

**Comment by ejgallego** [comment]
Ok, should be ready to merge .

We took a pragmatic solution here, we hope it doesn't cause much trouble to @fajb ; tried to find a very close style to the original PR but preserving `begin/end` and avoiding other style problems.

File: .ocamlformat-ignore [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+plugins/nsatz/*
+plugins/omega/*
+plugins/rtauto/*
+plugins/setoid/*
+plugins/ing/*
+plugins/setoid_ring/*
+plugins/ssr/*
+plugins/ssrmatching/*
+plugins/syntax/*
+# Enabled: none for now
```
Comment (Zimmi48) [inline .ocamlformat-ignore]: This comment should be updated.

**Comment by Zimmi48** [comment]
@ejgallego Currently, there are some files in the micromega code base that are checked to match versions extracted from Coq. These files should either be off-limit to micromega or the test should be updated to run ocamlformat on them before diffing.

File: .ocamlformat-ignore (hunk 1 above)
Comment (ejgallego) [reply .ocamlformat-ignore]: Fixed, thanks; also fixed the test suite failure [`micromega.ml` is extracted and must be ignored]

**Comment by ejgallego** [comment]
> @ejgallego Currently, there are some files in the micromega code base that are checked to match versions extracted from Coq. These files should either be off-limit to micromega or the test should be updated to run ocamlformat on them before diffing.

I've chosen to ignore them.

**Comment by Zimmi48** [comment]
> I've chosen to ignore them.

What do you mean by that? The test-suite is broken, so something needs to be done.

**Comment by ejgallego** [comment]
> What do you mean by that? The test-suite is broken, so something needs to be done.

I had already fixed this problem in my tree, just pushed.

**Comment by Zimmi48** [comment]
I see, thank you!

**Review by Zimmi48 — APPROVED** [review]

**Pull request merged**